transformer_engine/common/recipe/__init__.py

```diff
@@ -174,17 +174,29 @@ def scaling_factor_compute(amax: Tensor,
     reduce_amax: bool = True
     fp8_dpa: bool = False
     fp8_mha: bool = False
+    expected_tensor_class: Optional[Type] = None
 
     def __post_init__(self) -> None:
         assert self.fp8_format != Format.E5M2, "Pure E5M2 training is not supported."
+        try:
+            from transformer_engine.pytorch.tensor.float8_tensor import Float8Tensor
```

Comment:
Not a fan of adding framework specific logic in the core lib at all, even if it is guarded with try-except. So far we haven't had to do this and probably shouldn't introduce it here unless a strong reason exists to do so.

How about we reverse the logic and instead of including tensor info in the recipe, we include some recipe info in the tensor, and that should achieve the same goal of being able to equate the match the two together, right?